client-ui/src/app/model/lineage-graph/leneage-graph-legend.models.ts

```diff
@@ -15,26 +15,31 @@
  */
 
 
-import { LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY } from './leneage-graph.models';
+import {LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY} from './leneage-graph.models';
 
 
-export interface LineageGraphLegend {
+export type LineageGraphLegend = {
   title: string;
   color: string;
   description: string;
 }
 
-export function getLineageGraphLegend(): LineageGraphLegend[] {
-  return [
-    {
+export type LineageGraphLegendKeys =
+  "Usage" |
+  "Lineage" |
+  "Impact"
+
+export const LineageGraphLegends: { [key in LineageGraphLegendKeys]: LineageGraphLegend } =
+  Object.freeze({
+    Usage: {
```

Comment:
What will happen with that field if you will decide to refactor LineageGraphLegendKeys and rename Usage to somethoing else i ncase of having LineageGraphLegendKeys as a type?

Reply:
It will stop compiling. This is the intention :) With `[key in LineageGraphLegendKeys]` you clearly define the property names allowed for the object. Basically `enum` semantics.